BaboViolent2/Code/Server.cpp

```diff
@@ -1353,32 +1354,34 @@ void Server::update(float delay)
 		}
 
 		// Open map
-		CString filename("main/maps/%s.bvm", mapTransfers[i].mapName.s);
-		FILE* fic = fopen(filename.s, "rb");
-
-		if (fic)
-		{
-			net_svcl_map_chunk chunk;
-
-			// Read in a chunk
-			fseek(fic, 250 * mapTransfers[i].chunkNum, SEEK_SET);
-			chunk.size = (unsigned short)fread(chunk.data, 1, 250, fic);
-
-			// Send chunk
-			bb_serverSend((char*)&chunk, sizeof(net_svcl_map_chunk), NET_SVCL_MAP_CHUNK, mapTransfers[i].uniqueClientID);
-
-			// Accumulate bytes sent
-			bytesSent += 250;
-
-			// If some data was sent, keep this transfer
-			if(chunk.size != 0) 
-			{
-				mapTransfers[i].chunkNum++;
-				temp.push_back(mapTransfers[i]);
-			}
-		}
-
-		fclose(fic);
+        if (mapTransfers[i].mapName != "") {
```

Comment:
I've added this additional if-check because server crashes if client sends a packet without map name inside. It is potentially bad. Anyone can crash you server using special formed packets, I faced this issue when testing.